sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/jsonExpressions.scala

```diff
@@ -468,11 +468,13 @@ case class JsonTuple(children: Seq[Expression])
     new GenericInternalRow(row) :: Nil
   }
 
+  private val optimizeStringCopy = SQLConf.get.getConf(SQLConf.OPTIMIZE_STRING_COPY_IN_JSON_TUPLE)
+
   private def copyCurrentStructure(generator: JsonGenerator, parser: JsonParser): Unit = {
     parser.getCurrentToken match {
       // if the user requests a string field it needs to be returned without enclosing
       // quotes which is accomplished via JsonGenerator.writeRaw instead of JsonGenerator.write
-      case JsonToken.VALUE_STRING if parser.hasTextCharacters =>
+      case JsonToken.VALUE_STRING if optimizeStringCopy && parser.hasTextCharacters =>
```

Comment:
I'd simply remove this optimization. correctness is critical.

Reply:
Initially, I thought of removing the optimization but just hesitated that it could impact on users who have small fields. ok, let's remove it.